server/issue_handler.go

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package server
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/mattermost/mattermost-mattermod/model"
+	"github.com/mattermost/mattermost-server/v5/mlog"
+)
+
+func (s *Server) issueEventHandler(w http.ResponseWriter, r *http.Request) {
+	ctx, cancel := context.WithTimeout(context.Background(), defaultRequestTimeout*time.Second)
+	defer cancel()
+
+	event, err := PullRequestEventFromJSON(r.Body)
+	if err != nil {
+		mlog.Error("could not parse issue event", mlog.Err(err))
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	parts := strings.Split(event.Issue.GetHTMLURL(), "/")
+	if len(parts) < 4 {
+		mlog.Error("incorrect pattern for issue url")
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	mlog.Info("handle issue event", mlog.String("repoUrl", *event.Issue.HTMLURL), mlog.String("Action", event.Action), mlog.Int("PRNumber", event.PRNumber))
+	issue, err := s.GetIssueFromGithub(ctx, parts[len(parts)-4], parts[len(parts)-3], event.Issue)
+	if err != nil {
+		mlog.Error("could not get the issue from GitHub", mlog.Err(err))
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	if err := s.checkIssueForChanges(ctx, issue); err != nil {
+		mlog.Error("could not check issue for changes", mlog.Err(err))
+		w.WriteHeader(http.StatusInternalServerError)
+	}
+}
+
+func (s *Server) checkIssueForChanges(ctx context.Context, issue *model.Issue) error {
+	oldIssue, err := s.Store.Issue().Get(issue.RepoOwner, issue.RepoName, issue.Number)
+	if err != nil {
+		return err
+	}
+
+	if oldIssue == nil {
+		_, err = s.Store.Issue().Save(issue)
+		return err
```

Comment:
bug :)

Reply:
How? 😅

Reply:
Lol, we are just directly returning from here unconditionally :)